langroid/agent/base.py

```diff
@@ -1102,12 +1115,17 @@ def get_oai_tool_calls_classes(self, msg: ChatDocument) -> List[ToolMessage]:
                     or you need to enable this agent to handle this fn-call.
                     """
                 )
+                if tool_name not in self.all_llm_tools_known:
```

Comment:
We should omit these two lines - it's messing up the intended logic -- i.e. if any tool is found valid, then all_errors will be set to False down below, and things will be fine. Or am I missing something?

Reply:
You're right; they should be dropped.